core/src/main/java/com/linecorp/armeria/client/endpoint/healthcheck/HealthCheckedEndpointGroup.java

```diff
@@ -123,7 +123,11 @@ public static HealthCheckedEndpointGroupBuilder builder(EndpointGroup delegate,
         updateCandidates(delegate.initialEndpointsFuture().join());
 
         // Wait until the initial health of all endpoints are determined.
-        contexts.values().forEach(ctx -> ctx.initialCheckFuture.join());
```

Comment:
Would be nice to add a test that previously failed - I think it's possible to do it deterministically by having a initial endpoints future that doesn't complete while adding endpoints.

Reply:
Ah, thanks! Let me make it in another PR.